src/asynk/async_worker.rs

```diff
@@ -0,0 +1,144 @@
+    fn default() -> Self {
+        SleepParams {
+            sleep_period: 5,
+            max_sleep_period: 15,
+            min_sleep_period: 5,
+            sleep_step: 5,
+        }
+    }
+}
+#[derive(TypedBuilder, Debug)]
+pub struct AsyncWorker<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
+{
+    #[builder(setter(into))]
+    pub queue: AsyncQueue<Tls>,
+    #[builder(setter(into))]
+    pub task_type: Option<String>,
+    #[builder(setter(into))]
+    pub sleep_params: SleepParams,
+    #[builder(setter(into))]
+    pub retention_mode: RetentionMode,
+}
+impl<Tls> AsyncWorker<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
+{
+    pub async fn run(&mut self, task: Task) {
+        let result = self.execute_task(task).await;
+        self.finalize_task(result).await
+    }
+    async fn execute_task(&mut self, task: Task) -> Result<Task, (Task, String)> {
+        let actual_task: Box<dyn AsyncRunnable> =
+            serde_json::from_value(task.metadata.clone()).unwrap();
+        let task_result = actual_task.run(&self.queue.pool).await;
+
+        match task_result {
+            Ok(()) => Ok(task),
+            Err(error) => Err((task, error.description)),
+        }
+    }
+    async fn finalize_task(&mut self, result: Result<Task, (Task, String)>) {
+        match self.retention_mode {
+            RetentionMode::KeepAll => {
+                match result {
+                    Ok(task) => self
+                        .queue
+                        .update_task_state(task, FangTaskState::Finished)
+                        .await
+                        .unwrap(),
+                    Err((task, error)) => self.queue.fail_task(task, &error).await.unwrap(),
+                };
+            }
+            RetentionMode::RemoveAll => {
+                match result {
+                    Ok(task) => self.queue.remove_task(task).await.unwrap(),
+                    Err((task, _error)) => self.queue.remove_task(task).await.unwrap(),
+                };
+            }
+            RetentionMode::RemoveFinished => match result {
+                Ok(task) => {
+                    self.queue.remove_task(task).await.unwrap();
+                }
+                Err((task, error)) => {
+                    self.queue.fail_task(task, &error).await.unwrap();
+                }
+            },
+        }
+    }
+    pub fn sleep(&mut self) {
+        self.sleep_params.maybe_increase_sleep_period();
+
+        thread::sleep(Duration::from_secs(self.sleep_params.sleep_period));
```

Comment:
⚠️  thread::sleep is blocking the thread.

other tasks won't be able to do anything

let's use https://docs.rs/tokio/latest/tokio/time/fn.sleep.html

Reply:
i will change that